src/components/spotlight/SpotlightTypes.tsx

```diff
@@ -0,0 +1,73 @@
+/*
+ * Echoes React
+ * Copyright (C) 2023-2025 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import { ReactNode } from 'react';
+import { Props as JoyrideProps, Step as JoyrideStep } from 'react-joyride';
+import { TextNode } from '~types/utils';
+
+type SpotlightModalPlacement = 'left' | 'right' | 'top' | 'bottom' | 'center';
+
+export type SpotlightStep = JoyrideStep & {
+  placement?: SpotlightModalPlacement;
+};
+
+export interface SpotlightProps extends Omit<JoyrideProps, 'steps'> {
+  /**
+   * The CSS class name to add to the target element when it becomes active - defaults to "active" (optional)
+   */
+  activeTargetClassName?: string;
+
+  /**
+   * The label for the Back button - defaults to "Back" (optional)
+   */
+  backLabel?: TextNode;
+
+  /**
+   * The label for the Close button - defaults to "Close" (optional)
+   */
+  closeLabel?: TextNode;
+
+  /**
+   * The label for the Next button - defaults to "Next" (optional)
+   */
+  nextLabel?: TextNode;
+
+  /**
+   * The label for the Skip button - defaults to "Skip" (optional)
+   */
+  skipLabel?: TextNode;
+
```

Comment:
We might want to make them `TextNodeOptional` if we want to also allow passing `false` or `null` when we don't want to have values. It depends on how these are used in the component.